tools/project_manager/project_quick_check.php

```diff
@@ -14,31 +14,70 @@
 
 // get data passed into the page
 $projectid = validate_projectID("projectid", @$_REQUEST['projectid'], true);
+$callback = urldecode(array_get($_REQUEST, "callback", ""));
+
+$result = [];
+
+if($callback && $projectid)
+{
+    $project = new Project($projectid);
+    if($project->user_can_do_quick_check())
+    {
+        $result = get_test_results($projectid);
+        // loop thorugh results looking for errors
+        $found_error = FALSE;
+        foreach($result as $function => $values)
+        {
+            if($result[$function]["status"] == _("Error"))
+            {
+                $found_error = TRUE;
+                break;
+            }
+        }
+
+        if(!$found_error)
+        {
+            metarefresh(0, $callback);
+        }
+    }
+}
 
 $title = _("Project Quick Check");
-$page_text = _("This page tests the project in an attempt to uncover some common errors.");
 
 output_header($title, NO_STATSBAR);
 
 echo "<h1>$title</h1>";
 
-echo "<p>$page_text</p>";
-
-// show the form
-echo "<form method='GET'>";
-echo "<table>";
-echo  "<tr>";
-echo   "<td>" . _("Project ID") . "</td>";
-echo   "<td><input name='projectid' type='text' value='$projectid' size='40' required></td>";
-echo  "</tr>";
-echo "</table>";
-echo "<input type='submit' value='Test'>";
-echo "</form>";
+if($callback)
+{
+    echo "<p>";
+    echo _("One or more errors were found when checking the project. Review the project quick check results, address any problems found, and try the project transition again.");
+    echo "</p>";
 
-// stop if no projectid was specified
-if(empty($projectid))
+    echo "<p>";
+    echo sprintf(_("If you feel this project quick check result is incorrect, you can <a href='%s'>continue with the transition</a>."), $callback);
+    echo "</p>";
```

Comment:
I can understand wanting to give them the opportunity to override, but I'd move the ability to override to the bottom of the list of errors. If it's at the top of the page, they're not really motivated to look. Or may click on it accidentally.

Perhaps put a link back to the project page here, instead? "Return to Project Page (link) to reload corrected text files."?

(This is my opinion, only and I'm open to other suggestions.)

Reply:
Given Michael's suggestion below, this might be moot but I will see.

Reply:
Understood.